subsys/bluetooth/audio/has_client.c

```diff
@@ -631,6 +875,38 @@ int bt_has_client_conn_get(const struct bt_has *has, struct bt_conn **conn)
 	return 0;
 }
 
+int bt_has_client_presets_read(struct bt_has *has, uint8_t start_index, uint8_t count)
+{
+	struct has_inst *inst = HAS_INST(has);
+	int err;
+
+	BT_DBG("conn %p start_index 0x%02x count %d", (void *)inst->conn, start_index, count);
+
+	if (!inst->conn) {
+		return -ENOTCONN;
+	}
+
+	if (atomic_test_bit(inst->flags, HAS_DISCOVER_IN_PROGRESS) ||
+	    atomic_test_and_set_bit(inst->flags, HAS_CP_OPERATION_IN_PROGRESS)) {
+		return -EBUSY;
+	}
+
+	if (start_index == BT_HAS_PRESET_INDEX_NONE) {
+		return -EINVAL;
+	}
+
+	if (count == 0u) {
+		return -EINVAL;
+	}
```

Comment:
Prefer to use `CHECKIF` for values that the caller should know are valid/invalid

```suggestion
	CHECKIF(start_index == BT_HAS_PRESET_INDEX_NONE) {
		return -EINVAL;
	}

	CHECKIF(count == 0u) {
		return -EINVAL;
	}
```